From a975525e549d7c7c9198d45d7a05804711a2306d Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:08:10 -0400
Subject: [tools] perf/ui/gtk: Add support to accumulated hist stat

Message-id: <1410181752-25631-246-git-send-email-jolsa@redhat.com>
Patchwork-id: 91879
O-Subject: [PATCH RHEL7.1 BZ1134356 245/307] perf ui/gtk: Add support to accumulated hist stat
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit b09955b2a3d5fd02ed31d279f8c0ac29b32abe83
Author: Namhyung Kim <namhyung@kernel.org>
Date: Wed Oct 30 16:15:23 2013 +0900

description
===========
Print accumulated stat of a hist entry if requested.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/ui/gtk/hists.c b/tools/perf/ui/gtk/hists.c
index 9d90683..7e5da4a 100644
--- a/tools/perf/ui/gtk/hists.c
+++ b/tools/perf/ui/gtk/hists.c
@@ -47,11 +47,26 @@ static int perf_gtk__hpp_color_##_type(struct perf_hpp_fmt *fmt __maybe_unused,
      __percent_color_snprintf, true);   \
 }
 
+#define __HPP_COLOR_ACC_PERCENT_FN(_type, _field)    \
+static u64 he_get_acc_##_field(struct hist_entry *he)    \
+{          \
+ return he->stat_acc->_field;      \
+}          \
+          \
+static int perf_gtk__hpp_color_##_type(struct perf_hpp_fmt *fmt __maybe_unused, \
+           struct perf_hpp *hpp,   \
+           struct hist_entry *he)   \
+{          \
+ return __hpp__fmt_acc(hpp, he, he_get_acc_##_field, " %6.2f%%",  \
+         __percent_color_snprintf, true);   \
+}
+
 __HPP_COLOR_PERCENT_FN(overhead, period)
 __HPP_COLOR_PERCENT_FN(overhead_sys, period_sys)
 __HPP_COLOR_PERCENT_FN(overhead_us, period_us)
 __HPP_COLOR_PERCENT_FN(overhead_guest_sys, period_guest_sys)
 __HPP_COLOR_PERCENT_FN(overhead_guest_us, period_guest_us)
+__HPP_COLOR_ACC_PERCENT_FN(overhead_acc, period)
 
 #undef __HPP_COLOR_PERCENT_FN
 
@@ -68,6 +83,8 @@ void perf_gtk__init_hpp(void)
     perf_gtk__hpp_color_overhead_guest_sys;
  perf_hpp__format[PERF_HPP__OVERHEAD_GUEST_US].color =
     perf_gtk__hpp_color_overhead_guest_us;
+ perf_hpp__format[PERF_HPP__OVERHEAD_ACC].color =
+    perf_gtk__hpp_color_overhead_acc;
 }
 
 static void callchain_list__sym_name(struct callchain_list *cl,
-- 
1.7.1